docs/decisions/0045-updated-vector-store-design.md

```diff
@@ -0,0 +1,810 @@
+    IAsyncEnumerable<string> ListCollectionNamesAsync(CancellationToken cancellationToken = default);
+    Task<bool> CollectionExistsAsync(string name, CancellationToken cancellationToken = default);
+    Task DeleteCollectionAsync(string name, CancellationToken cancellationToken = default);
+}
+
+// DB Specific Update implementations
+class AzureAISearchMemoryCollectionUpdateService: IMemoryCollectionUpdateService;
+class RedisMemoryCollectionUpdateService: IMemoryCollectionUpdateService;
+
+// Combined Create + Update Interface
+interface IMemoryCollectionService: IMemoryCollectionCreateService, IMemoryCollectionUpdateService {}
+
+// Base abstract class that forwards non-create operations to provided service.
+abstract class MemoryCollectionService(IMemoryCollectionUpdateService collectionsUpdateService): IMemoryCollectionService
+{
+    public abstract Task CreateCollectionAsync(string name, CancellationToken cancellationToken = default);
+    public IAsyncEnumerable<string> ListCollectionNamesAsync(CancellationToken cancellationToken = default) { return collectionsUpdateService.ListCollectionNamesAsync(cancellationToken); }
+    public Task<bool> CollectionExistsAsync(string name, CancellationToken cancellationToken = default) { return collectionsUpdateService.CollectionExistsAsync(name, cancellationToken); }
+    public Task DeleteCollectionAsync(string name, CancellationToken cancellationToken = default) { return collectionsUpdateService.DeleteCollectionAsync(name, cancellationToken); }
+}
+
+// Collections service implementations, that inherit from base class, and just adds the different creation implementations.
+class AzureAISearchChatHistoryMemoryService(AzureAISearchMemoryCollectionUpdateService updateService): MemoryCollectionService(updateService);
+class AzureAISearchSemanticCacheMemoryService(AzureAISearchMemoryCollectionUpdateService updateService): MemoryCollectionService(updateService);
+class AzureAISearchMLIndexMemoryService(AzureAISearchMemoryCollectionUpdateService updateService): MemoryCollectionService(updateService);
+
+// Customer collections service implementation, that uses the base Azure AI Search implementation for get, doesExist and delete, but adds it's own creation.
+class ContosoProductsMemoryCollectionService(AzureAISearchMemoryCollectionUpdateService updateService): MemoryCollectionService(updateService);
+
+```
+
+#### Option 5 - Separated collection and record management with collection create separate from other operations, with overall aggregation class on top.
+
+Same as option 3 / 4, plus:
+
+```cs
+
+interface IMemoryService : IMemoryCollectionCreateService, IMemoryCollectionService, IMemoryRecordService
+{    
+}
+
+// Create a static factory that produces one of these, so only the interface is public, not the class.
+internal class CombinedMemoryService<TDataModel>(IMemoryCollectionCreateService creation, IMemoryCollectionService collections, IMemoryRecordService<TDataModel> records): IMemoryService
```

Comment:
Is `IMemoryCollectionService` already a combination of `IMemoryCollectionCreateService` and `IMemoryCollectionUpdateService`?

Reply:
Yeah, that's the idea.  What I'm illustrating here is the ability to create a class that combines the two in a way that makes it easy to just add the bits that are different.  E.g. you grab the update piece from the library, but add your own create implementation.